qdrant_client/uploader/grpc_uploader.py

```diff
@@ -0,0 +1,113 @@
+import asyncio
+from itertools import count
+from typing import List, Iterable, Any
+
+from grpclib.client import Channel
+
+from qdrant_client import grpc
+from qdrant_client.grpc import PointsStub, PointStruct, PointId, KeywordPayload, IntegerPayload, FloatPayload, \
+    GeoPayload
+from qdrant_client.uploader.uploader import BaseUploader
+
+
+def iter_over_async(ait, loop):
+    ait = ait.__aiter__()
+
+    async def get_next():
+        try:
+            obj = await ait.__anext__()
+            return False, obj
+        except StopAsyncIteration:
+            return True, None
+
+    while True:
+        done, obj = loop.run_until_complete(get_next())
+        if done:
+            break
+        yield obj
+
+
+def process_payload(payload: dict):
+    res = {}
+    for key, val in payload.items():
+        if isinstance(val, str):
+            res[key] = grpc.Payload(keyword=KeywordPayload(values=[val]))
+            continue
+
+        if isinstance(val, int):
+            res[key] = grpc.Payload(integer=IntegerPayload(values=[val]))
+            continue
+
+        if isinstance(val, float):
+            res[key] = grpc.Payload(float=FloatPayload(values=[val]))
+            continue
+
+        if isinstance(val, dict):
+            if 'lon' in val and 'lat' in val:
+                res[key] = grpc.Payload(geo=GeoPayload(values=[grpc.GeoPoint(lat=val['lat'], lon=val['lon'])]))
+                continue
+            else:
+                raise RuntimeWarning(f"Unsupported payload: ({key} -> {val})")
+
+        if isinstance(val, list):
+            if all(isinstance(v, str) for v in val):
+                res[key] = grpc.Payload(keyword=KeywordPayload(values=val))
+                continue
+
+            if all(isinstance(v, int) for v in val):
+                res[key] = grpc.Payload(integer=IntegerPayload(values=val))
+                continue
+
+            if all(isinstance(v, float) for v in val):
+                res[key] = grpc.Payload(float=FloatPayload(values=val))
+                continue
+
+            if all(isinstance(v, dict) and 'lon' in v and 'lat' in v for v in val):
+                res[key] = grpc.Payload(geo=GeoPayload(
+                    values=[grpc.GeoPoint(lat=v['lat'], lon=v['lon']) for v in val],
+                ))
+                continue
+        raise RuntimeError(f"Payload {key} have unsupported type {type(val)}")
+    return res
+
+
+async def upload_batch_grpc(service: PointsStub, collection_name: str, batch):
```

Comment:
In GRPC context, the name `service` is usually used for server-side implementations, and instances of `XxxStub` are usually named `xxx_client`. This is just a nit, but I believe it's better to stick to the convention.